gitlab/util.go

```diff
@@ -143,6 +143,21 @@ func stringToMergeMethod(s string) *gitlab.MergeMethodValue {
 	return &value
 }
 
+func stringToSquashOptionValue(s string) *gitlab.SquashOptionValue {
+	lookup := map[string]gitlab.SquashOptionValue{
+		"never":       gitlab.SquashOptionNever,
+		"always":      gitlab.SquashOptionAlways,
+		"default_on":  gitlab.SquashOptionDefaultOn,
+		"default_off": gitlab.SquashOptionDefaultOff,
+	}
+
+	value, ok := lookup[s]
```

Comment:
We should error here.

Reply:
there is error handling on line 155, or do you mean something else?

Reply:
But this causes places like this:

```golang
		SquashOption:                              stringToSquashOptionValue(d.Get("squash_option").(string)),
```

to have issues with number of returned agruments, it exactly expects one of them, I am not sure how should I handle that.

Reply:
I'd be happy to follow some example on how to implement stringMapping function as validation but it seems like that this one does not differ from the others :)